### PR TITLE
fix: only add aria-activedescendant when menu is open

### LINKED ISSUE #921: [useSelect] aria-activedescendant does not have valid value with controlled selectedItem
- `downshift` version: 4.0.8
- `node` version: 12.14.1
- `npm` (or `yarn`) version: yarn 1.21.1

**Relevant code or config**

```javascript
import React from 'react'
import { render } from 'react-dom'
import { useSelect } from 'downshift'

const items = ['Apples', 'Bananas', 'Oranges']

const Select = ({ items, selectedItem }) => {
  const {
    getItemProps,
    getLabelProps,
    getMenuProps,
    getToggleButtonProps,
    isOpen,
  } = useSelect({ items, selectedItem })

  return (
    <div>
      <label {...getLabelProps()}>Fruit: </label>
      <button {...getToggleButtonProps()}>{selectedItem}</button>
      {/* <ul {...getMenuProps(isOpen ? undefined : { "aria-activedescendant": null })}> */}
      <ul {...getMenuProps()}>
        {isOpen &&
          items.map((item, index) => (
            <li key={item} {...getItemProps({ item, index })}>
              {item}
            </li>
          ))}
      </ul>
    </div>
  )
}

const App = () => <Select items={items} selectedItem={items[0]} />

render(<App />, document.getElementById('root'))
```

**What you did**:
use `useSelect` with controlled props (`items` and `selectedItem`)

**What happened**:
when used with controlled props, `useSelect`'s `getMenuProps` ouputs an `aria-activedescendant` attribute pointing to a non-existing id (because the `<li>` elements are only rendered when the select is open.
Lighthouse therefore complains with "[aria-*] attributes do not have valid values".

**Reproduction repository**:

**Problem description**:

**Suggested solution**:
I'm currently working aroung this with `{...getMenuProps(isOpen ? undefined : { "aria-activedescendant": null })}` (see above), but i guess there is a better solution?

### PR DESCRIPTION
**Pull request by stefanprobst**
**What**: in `useSelect`/`useCombobox` only add `aria-activedescendant` attribute when menu is open to avoid pointing to non-existing id.

**Why**: Fixes #921

**How**: check if `isOpen` is `true` before adding `aria-activedescendant`.

**Checklist**:

- [ ] Documentation
- [x] Tests
- [ ] TypeScript Types
- [ ] Flow Types
- [x] Ready to be merged

### CODE REVIEW

File: src/hooks/useCombobox/__tests__/getToggleButtonProps.test.js [hunk 1]
```diff
@@ -208,6 +208,8 @@ describe('getToggleButtonProps', () => {
         )
 
         fireEvent.click(toggleButton)
+        // fireEvent.blur(input)
```
Comment (silviuavram) [inline src/hooks/useCombobox/__tests__/getToggleButtonProps.test.js]: Can we remove this?

Comment (stefanprobst) [reply src/hooks/useCombobox/__tests__/getToggleButtonProps.test.js]: sure -- comment was just for symmetry with the comments above

Comment (silviuavram) [reply src/hooks/useCombobox/__tests__/getToggleButtonProps.test.js]: I mean also the line below, I don't see why is it needed.

Comment (stefanprobst) [reply src/hooks/useCombobox/__tests__/getToggleButtonProps.test.js]: the `input.blur()` is needed actually (or `toggleButton.focus()` alternatively). i had assumed it was just an oversight this test did not have that line while e.g. the one above did.
without manually blurring/focusing the actions fire in wrong order:

without manual blur:
* `__togglebutton_click__`
* `__togglebutton_click__`
* `__togglebutton_click__`
* `__input_blur__`

with:
* `__togglebutton_click__`
* `__togglebutton_click__`
* `__input_blur__`
* `__togglebutton_click__`

**Comment by silviuavram** [comment]
As I merged a big change as part of v5 I re-created this PR. Ran the tests without the `.blur` in the discussion and they were passing. Please take a look at https://github.com/downshift-js/downshift/pull/945

**Pull request closed without merging**